scalameta/io/shared/src/main/scala/scala/meta/io/Fragment.scala

```diff
@@ -0,0 +1,16 @@
+package scala.meta
+package io
+
+import java.net._
+import java.io._
+import org.scalameta.data._
+
+@data class Fragment(base: AbsolutePath, name: RelativePath) {
+  def uri: URI = {
+    val suri = if (base.toString.endsWith(".jar")) s"jar:file:$base!/$name" else s"file:$base/$name"
```

Comment:
Yes, it does. However, this code contained other problems that should be fixed now.